### PR TITLE
AUTOTOOLS: minor updates for configure scripts

### PR DESCRIPTION
**Pull request by shamisp**
* Help message update for knem
* XPMEM autodetect code

Signed-off-by: Pavel Shamis (Pasha) <pasharesearch@gmail.com>

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2604/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1961/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2605/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1962/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2606/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1963/ for details (Mellanox internal link).

**Comment by MattBBaker** [comment]
The Oak Ridge failure looks like like the Mlx one.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2607/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1964/ for details (Mellanox internal link).

**Comment by shamisp** [comment]
Fixing #1060

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2608/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1965/ for details (Mellanox internal link).

File: config/m4/xpmem.m4 [hunk 1]
```diff
@@ -3,28 +3,35 @@
...
-                             AC_DEFINE([HAVE_XPMEM], [1], [Enable the use of XPMEM])
-                             transports="${transports},xpmem"
-                             xpmem_happy="yes"],
-                             [AC_MSG_WARN([XPMEM requested but not found])
-                             AC_DEFINE([HAVE_XPMEM], [0], [Disable the use of XPMEM])])])],
-      [AC_MSG_WARN([XPMEM was explicitly disabled])
-       AC_DEFINE([HAVE_XPMEM], [0], [Disable the use of XPMEM])]
-)
+      [AS_IF([test ! -d $with_xpmem],
+             [AC_MSG_NOTICE([XPMEM - faild to open the requested location ($with_xpmem), guessing ...])
```
Comment (miked-mellanox) [inline config/m4/xpmem.m4]: s/faild/failed

**Comment by MrBr-github** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2613/ for details.

**Comment by MrBr-github** [comment]
bot:bgate:rebuild

**Comment by MrBr-github** [comment]
bot:bgate:rebuild

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr__NEW/2/ for details.

**Comment by MrBr-github** [comment]
bot:bgate:rebuild

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1967/ for details (Mellanox internal link).

File: config/m4/xpmem.m4 (hunk 1 above)
Comment (shamisp) [inline config/m4/xpmem.m4]: 10x

File: config/m4/xpmem.m4 [hunk 2]
```diff
@@ -3,28 +3,35 @@
...
+             # Check for a Cray module
+             PKG_CHECK_MODULES([CRAY_XPMEM], [cray-xpmem],
+                               [transports="${transports},xpmem"
+                                xpmem_happy=yes
+                                AC_SUBST(XPMEM_CPPFLAGS,  "$CRAY_XPMEM_CFLAGS")
+                                AC_SUBST(XPMEM_LDFLAGS,   "$CRAY_XPMEM_LIBS")
+                                AC_DEFINE([HAVE_XPMEM], [1], [Enable the use of XPMEM])],
+                                # If Cray module failed try to search
+                                [with_xpmem=$(find /opt/xpmem /usr/local -name xpmem.h 2>/dev/null |
```
Comment (shamisp) [inline config/m4/xpmem.m4]: valid point

**Comment by MrBr-github** [comment]
bot:bgate:rebuild

**Comment by MrBr-github** [comment]
bot:bgateNEW:rebuild

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr__NEW/22/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2623/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1976/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
bot:bgateNEW:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr__NEW/44/ for details.

**Pull request merged**